springapp/src/main/java/com/javaacademy/robot/repository/BookRepository.java

```diff
@@ -1,11 +1,24 @@
 package com.javaacademy.robot.repository;
 
 import com.javaacademy.robot.model.Book;
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.data.repository.PagingAndSortingRepository;
 import org.springframework.stereotype.Component;
 import org.springframework.transaction.annotation.Transactional;
 
+import java.util.List;
+
 @Transactional
 @Component
 public interface BookRepository extends PagingAndSortingRepository<Book, Long> {
+    @Query("select b from Book b " +
+            "join b.authors a " +
+            "join b.categories c " +
+            "where lower(b.title) like %?1% and " +
+            "lower(a) like %?2% and " +
+            "lower(c) like %?3% and " +
+            "lower(b.shopName) like %?4% and " +
+            "b.retailPriceAmount >= ?5 and " +
+            "b.retailPriceAmount <= ?6")
```

Comment:
Why 5 and 6 ???

Reply:
These are parameter numbers, they are marked with `?`. :P